fe/fe-core/src/main/java/com/starrocks/catalog/Type.java

```diff
@@ -1497,7 +1499,27 @@ public Type getResultType() {
         }
     }
 
-    public static Type getCmpType(Type t1, Type t2) {
+    public static Type getCmpType(Type t1, Type t2, boolean isBetween) {
+        // if predicate is 'IN' and one type is string ,another type is not float
+        // we choose string or decimal as cmpType according to session variable cboEqBaseType
+        if (!isBetween &&
```

Comment:
what the behavior when varchar_col in (int_value, string_val, double_val)?

Reply:
(varchar, int) -> varchar, (varchar, string)->varchar, (varchar,double)->double, so all cast into double